diff --git a/source/2nd_loader/src/rdk_ddr_api.c b/source/2nd_loader/src/rdk_ddr_api.c
index 775581e..88f5fbe 100644
--- a/source/2nd_loader/src/rdk_ddr_api.c
+++ b/source/2nd_loader/src/rdk_ddr_api.c
@@ -176,8 +176,8 @@ void MMC_EnableSignal_dfi_init_complete(void)
 
 void MMC_SetBackRegisters(void)
 {
-    MMC_WriteReg(0x0060, ((MMC_ReadReg(0x0060) & 0xFFFFFFFE) | 0x00000000));
-    MMC_WriteReg(0x0030, ((MMC_ReadReg(0x0030) & 0xFFFFFFDC) | 0x00000000));
+    MMC_WriteReg(0x0060, 0x00000000);            /* Setting up 1chip sim environment */
+    MMC_WriteReg(0x0030, 0x00000108);            /* Setting up 1chip sim environment */
 }
 
 int32_t MMC_WaitBitForSW_Done(void)
