From 69d9244d4b1db5bb51b530c885bf4cdbe9d0bf0b Mon Sep 17 00:00:00 2001
From: Canh Dao <canh.dao.ct@renesas.com>
Date: Mon, 9 Aug 2021 14:08:51 +0700
Subject: [PATCH] ptrace rzv2m

Signed-off-by: Canh Dao <canh.dao.ct@renesas.com>
---
 arch/arm64/include/asm/ptrace.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/include/asm/ptrace.h b/arch/arm64/include/asm/ptrace.h
index 163970d..68aa9a0 100644
--- a/arch/arm64/include/asm/ptrace.h
+++ b/arch/arm64/include/asm/ptrace.h
@@ -24,6 +24,7 @@
 /* Current Exception Level values, as contained in CurrentEL */
 #define CurrentEL_EL1		(1 << 2)
 #define CurrentEL_EL2		(2 << 2)
+#define CurrentEL_EL3           (3 << 2)
 
 /* AArch32-specific ptrace requests */
 #define COMPAT_PTRACE_GETREGS		12
-- 
2.7.4

