diff --git a/drivers/spi/rzv2m_csi.c b/drivers/spi/rzv2m_csi.c
index 4f01c7b..4625bbd 100644
--- a/drivers/spi/rzv2m_csi.c
+++ b/drivers/spi/rzv2m_csi.c
@@ -545,7 +545,7 @@ static int rzv2m_csi_set_fifo_trg(struct spi_master *master, struct spi_transfer
 	const void *tx_buf = t->tx_buf;
 	void *rx_buf = t->rx_buf;
     
-	if(!tx_buf){
+	if(tx_buf != NULL){
 		if(is_16bit_data_leng(p)){ //16 bit mode
 			if((info->transmit_trigger_level > 4)){
 				ret = -EINVAL;
@@ -563,7 +563,7 @@ static int rzv2m_csi_set_fifo_trg(struct spi_master *master, struct spi_transfer
 		}		
 	}
 	
-	if(!rx_buf){
+	if(rx_buf != NULL){
 		//setting for receive trigger level
 		if(is_16bit_data_leng(p)){ //16 bit mode
 			if((info->receive_trigger_level > 4)){
